### DISCUSSION TITLE
Golang not supported

### DISCUSSION THREAD

**Discussion by igorskyflyer** [anuraghazra/github-readme-stats#2942] [Q&A] [answered]
### Describe the bug

Is the Go (Golang) programming language not supported by the Top languages card?

Because I have this card in my README:

![igorskyflyer's Top languages](https://github-readme-stats-igordvlpr.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&count_private=true&langs_count=20&hide_border=true&cache_seconds=1800&custom_title=Top%20Languages)

And all the languages appear except Go, I have a repo with Go code, available here [Go in my repos](https://github.com/igorskyflyer?tab=repositories&q=&type=&language=go).

P.S. Rust is detected even though I have more Go code than Rust.

### Expected behavior

Go should be rendered on the Top languages card.

### Screenshots / Live demo link

https://github-readme-stats-igordvlpr.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&count_private=true&langs_count=20&hide_border=true&cache_seconds=1800&custom_title=Top%20Languages

**Comment by rickstaa** [comment]
@igorskyflyer I quickly checked your profile, and GoLang does show up for me. I'm not sure what goes wrong in your 

```md
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&custom_title=Top&Languages&cache_seconds=1800)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&custom_title=Top&Languages&cache_seconds=1800)](https://github.com/anuraghazra/github-readme-stats)

**BTW**: The `count_private` flag has been deprecated (see https://github.com/anuraghazra/github-readme-stats/pull/2736). Additionally, for the public instance, the minimum `cache_seconds` is `14400` (see https://github.com/anuraghazra/github-readme-stats#common-options). For private Vercel instances, we do have the `CACHE_SECONDS` environment variable to bypass this limit (see https://github.com/anuraghazra/github-readme-stats#disable-rate-limit-protections).

**Reply by igorskyflyer** [reply]
Thanks for the input, about the flags, I am aware that `cache_seconds` has a minimum of `14400` seconds I was testing that some time ago but never bothered to change it to the minimum but that being said, that shouldn't even matter because my repo with the Go code is 2 days old now, so any caching to prevent exceeding the rate limit should have been long gone be now (14400s = 4h). About the `count_private` flag, I didn't see that API change, thanks for the heads up! 😁

And I agree, it's weird that the original deploy renders Go while my own doesn't 😵

**Reply by rickstaa** [reply]
If your fork is up to date, it should show the same results as the public fork (See https://github.com/anuraghazra/github-readme-stats#keep-your-fork-up-to-date) 🤔.

**My fork**

```md
[![Top Langs](https://github-readme-stats-git-master-rickstaa.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&custom_title=Top&Languages&cache_seconds=1800)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats-git-master-rickstaa.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&custom_title=Top&Languages&cache_seconds=1800)](https://github.com/anuraghazra/github-readme-stats)

**Reply by igorskyflyer** [reply]
I forced a pull like 20 minutes ago, even though I already synced the repos yesterday, when I saw that Go wasn't detected. We will see in a few hours I guess. 😅

**Reply by rickstaa** [reply]
Cool, feel free to accept my answer if it works 👍🏻. You can also use the `CACHE_SECONDS` env variable to speed up the cache period in your Vercel instance.

**Reply by rickstaa** [reply, marked as answer]
I checked your instance, and it still doesn't show GoLang. From my side, there is not much I can do since I cannot reproduce the issue on my instance and have no access to your PAT (!!Please don't share it!!). The best thing you can do is:

1. Read the https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md guidelines on how you can debug this package.
2. Place a Vscode breakpoint on:

https://github.com/anuraghazra/github-readme-stats/blob/02ebd3243b4dc1aba224c7c75c23ebd3e4867ed2/api/top-langs.js#L45-L49

3. Start the debugger and check whether the data returned by GitHub contains `GoLang` entries

Since Vercel uses the PAT you set in the web interface, it is like debugging on the server. If you get the `GoLang` results returned, and they do not show up on the card, there is a bug in our code, and you can open a bug report. If you don't get `GoLang` results returned, your PAT token is incorrect, or there is a bug in the GraphQL endpoint. In that case, you'll need to contact GitHub.

If it does contain GoLang entries
#### Public instance

![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&cache_seconds=14400&custom_title=Top%20Languages&disable_animations=true)

```md
![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&cache_seconds=14400&custom_title=Top%20Languages&disable_animations=true)
```

#### Your instance

![Top Langs](https://github-readme-stats-nu-ecru.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&cache_seconds=14400&custom_title=Top%20Languages&disable_animations=true)

```md
![Top Langs](https://github-readme-stats-nu-ecru.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&theme=tokyonight&langs_count=20&hide_border=true&cache_seconds=14400&custom_title=Top%20Languages&disable_animations=true)
```

```md
![Top Langs](https://github-readme-stats-nu-ecru.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&langs_count=20&random=123324234)
```

![Top Langs](https://github-readme-stats-nu-ecru.vercel.app/api/top-langs/?username=igorskyflyer&layout=compact&langs_count=20&random=123324234)

**Reply by igorskyflyer** [reply]
Actually, yesterday night, I purged all the cache on my Vercel instance and when that didn't help I purged the caching on GitHub's server as well (`camo.githubusercontent.com`) and I saw my Vercel instance generate a brand new card, yet, it's the card we both see, without Go, yet again.

Thanks for today's input, I guess the issue is with GitHub's API itself, since my PAT is correct because PHP and C# are counted and they are in my private repos, so we can dismiss that possibility.

Anyway, here's the screenshot of my debug session, no Golang being retrieved...

<div align="center">
	<img src="https://github.com/anuraghazra/github-readme-stats/assets/20957750/11660930-6b89-4c25-a4f5-f49a6682504b" alt="igorskyflyer's Top Languages card">
</div>

Either way, thanks for all the help Rick and enjoy your Sunday, mate! 🍻

**Reply by igorskyflyer** [reply]
Also, just went even further, I think the problem is here - though I don't do GraphQL.

<div align="center">
	<img src="https://github.com/anuraghazra/github-readme-stats/assets/20957750/e4b1dd86-4354-4885-8f0f-e791362c2a41">
</div>

We only fetch first 100 repos? Does that query param mean that? If that's the case, then that's causing this because I do have 100+ repos, 108 to be exact (private + public).
And the 1st repo with Golang is a new one, meaning it's beyond the limit of 100 repos fetched, i.e. any new repos I create, their languages will be ignored, right?
And this is also the reason why the public API shows Golang! Because it cannot access my private repos, it doesn't hit the 100 limit, so it fetches the new repos as well.

When I tried to increase the param, GitHub complained it can only fetch the first 100. 😐

And related to an issue you actually opened in the past #1852.

**Reply by rickstaa** [reply]
@igorskyflyer Ah, I forgot about the only 100 repos issue. This will be fixed when we release the [GitHub action](https://github.com/anuraghazra/github-readme-stats/issues/2179). Thanks for debugging the problem 👍🏻.
